src/net.h

```diff
@@ -318,13 +320,20 @@ class CConnman
     void SetTryNewOutboundPeer(bool flag);
     bool GetTryNewOutboundPeer();
 
+    void StartExtraBlockRelayPeers() {
```

Comment:
Have you considered a `MarkExtraBlockRelayPeers` function which a) set `m_start_extra_block_relay_peers` to true if isn't and b) return the state of `m_start_extra_block_relay_peers` ? That way you can drop new `m_initial_sync_finished` I think

Reply:
I had a slight preference for not frequently looking at an atomic; `m_initial_sync_finished` is only used in one thread and therefore doesn't need locks.  This issue seems minor either way, so planning to leave it as-is.